src/micm/micm.cpp

```diff
@@ -1,180 +1,207 @@
-/**
- * This file contains the implementation of the MICM class, which represents a multi-component
- * reactive transport model. It also includes functions for creating and deleting MICM instances,
- * Copyright (C) 2023-2024 National Center for Atmospheric Research,
- *
- * SPDX-License-Identifier: Apache-2.0* creating solvers, and solving the model.
- */
+// Copyright (C) 2023-2024 National Center for Atmospheric Research,
+// SPDX-License-Identifier: Apache-2.0
+//
+// This file contains the implementation of the MICM class, which represents a
+// multi-component reactive transport model. It also includes functions for
+// creating and deleting MICM instances, creating solvers, and solving the model.
+#include <cmath>
 #include <filesystem>
 #include <iostream>
-#include <cmath>
 
 #include <micm/solver/rosenbrock_solver_parameters.hpp>
 #include <micm/system/species.hpp>
+#include <musica/error.hpp>
 #include <musica/micm.hpp>
 
-MICM *create_micm(const char *config_path, int *error_code)
-{
-    try
-    {
-        MICM *micm = new MICM();
-        *error_code = micm->create_solver(std::string(config_path));
-        return micm;
-    }
-    catch (const std::bad_alloc &e)
-    {
-        *error_code = 1;
-        return nullptr;
-    }
+MICM *create_micm(const char *config_path) {
+  MICM *micm = new MICM();
+  try {
+    micm->create_solver(std::string(config_path));
+    return micm;
+  } catch (const std::exception &e) {
+    delete micm;
+    Error(909039518, std::string("Error creating MICM solver: " + std::string(e.what())).c_str());
```

Comment:
Since these codes are really meant to make where the error happened identifiable, maybe this is another time where a macro is useful and we use `__LINE__` and `__FILE__` to do this instead?